src/App.jsx

```diff
@@ -6,22 +6,30 @@ import './App.css'
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function App() {
 
-const useSemiPersistentState = () => {
-  const [todoList,setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
+  const [todoList,setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect (() => { 
+     const newPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({data:{todoList: JSON.parse(localStorage.getItem('savedTodoList')) || []}})
```

Comment:
in this code data: {
todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
}, you assign to resolve.data todoList: JSON.parse(localStorage.getItem("savedTodoList")), you shouldn't use || [] here, because useState here  const [todoList, setTodoList] = React.useState([]); assign empty array as todoList, and in the other useEffect (lines 27-31) you are creating in localStorage variable savedTodoList  and set it to todoList: localStorage.setItem('savedTodoList', JSON.stringify(todoList)).